scene/2d/tile_map_layer.cpp

```diff
@@ -249,7 +248,7 @@ void TileMapLayer::_rendering_update(bool p_force_cleanup) {
 
 	if (!forced_cleanup) {
 		// List all quadrants to update, recreating them if needed.
-		if (dirty.flags[DIRTY_FLAGS_TILE_SET] || dirty.flags[DIRTY_FLAGS_LAYER_IN_TREE] || _rendering_was_cleaned_up) {
+		if (dirty.flags[DIRTY_FLAGS_TILE_SET] || dirty.flags[DIRTY_FLAGS_LAYER_IN_TREE] || dirty.flags[DIRTY_FLAGS_LAYER_HIGHLIGHT_MODE] || _rendering_was_cleaned_up) {
```

Comment:
@groud AFAICT this will result in recreating all the quadrants' items on highlight mode change. Shouldn't this flag be checked and items' `self_modulate` updated here instead:
https://github.com/godotengine/godot/blob/149a4b4ca12e2baf700020fc10f00d5b0a5e9f0d/scene/2d/tile_map_layer.cpp#L415-L429
?

Reply:
Oh yeah that would be better, you're right